src/PowerShellEditorServices/Services/TextDocument/Handlers/TextDocumentHandler.cs

```diff
@@ -59,30 +61,34 @@ public override Task<Unit> Handle(DidChangeTextDocumentParams notification, Canc
             return Unit.Task;
         }
 
-        protected override TextDocumentSyncRegistrationOptions CreateRegistrationOptions(SynchronizationCapability capability, ClientCapabilities clientCapabilities) => new()
+        protected override TextDocumentSyncRegistrationOptions CreateRegistrationOptions(SynchronizationCapability capability, ClientCapabilities clientCapabilities)
         {
-            DocumentSelector = LspUtils.PowerShellDocumentSelector,
-            Change = Change,
-            Save = new SaveOptions { IncludeText = true }
-        };
+            _isFileWatcherSupported = clientCapabilities.Workspace.DidChangeWatchedFiles.IsSupported;
+            return new TextDocumentSyncRegistrationOptions()
+            {
+                DocumentSelector = LspUtils.PowerShellDocumentSelector,
+                Change = Change,
+                Save = new SaveOptions { IncludeText = true }
+            };
+        }
 
         public override Task<Unit> Handle(DidOpenTextDocumentParams notification, CancellationToken token)
         {
+            // We use a fake Uri because we only want to test the LanguageId here and not if the
+            // file ends in ps*1.
+            TextDocumentAttributes attributes = new(s_fakeUri, notification.TextDocument.LanguageId);
+            if (!LspUtils.PowerShellDocumentSelector.IsMatch(attributes))
+            {
+                return Unit.Task;
+            }
+
```

Comment:
I changed this up a bit because it seems like the previous behavior was wrong. We were storing files we get `DidOpenTextDocument` notifications for, even if they weren't for PowerShell files. I don't think we would receive notifications for anything else, but if we did it doesn't make sense to store it.

Reply:
Ah, got it.